### DISCUSSION TITLE
Help with Angular 9 and PrimeNG

### DISCUSSION THREAD

**Discussion by eduardoolimpiodev** [primefaces/community#261] [PrimeNG]
Hello everyone, I hope you are well.
Recently, I've been updating an Angular project from version 2.0 to higher versions, going from 2 to 4, 4 to 5, 5 to 6, and so on... Everything was working fine until I updated my project to Angular 9, at which point my code broke. I would appreciate your help! Thanks.

My Package.json:
`{
  "name": "weblims-ng4",
  "version": "5.0.0",
  "license": "MIT",
  "scripts": {
    "ng": "ng",
    "start": "ng serve",
    "build": "ng build",
    "buildHOM": "ng build --prod --env=hom",
    "test": "ng test",
    "lint": "ng lint",
    "e2e": "ng e2e"
  },
  "private": true,
  "dependencies": {
    "@angular/animations": "^9.1.13",
    "@angular/cdk": "^9.2.4",
    "@angular/cli": "^9.1.15",
    "@angular/common": "^9.1.13",
    "@angular/compiler": "^9.1.13",
    "@angular/core": "^9.1.13",
    "@angular/forms": "^9.1.13",
    "@angular/material": "^9.2.4",
    "@angular/platform-browser": "^9.1.13",
    "@angular/platform-browser-dynamic": "^9.1.13",
    "@angular/platform-server": "^9.1.13",
    "@angular/router": "^9.1.13",
    "@types/minimatch": "^3.0.5",
    "chart.js": "2.1.3",
    "classlist.js": "^1.1.20150312",
    "core-js": "^2.4.1",
    "font-awesome": "^4.7.0",
    "fullcalendar": "^3.1.0",
    "jquery": "^3.1.1",
    "jsbarcode": "^3.5.8",
    "moment": "^2.17.1",
    "nanoscroller": "0.8.7",
    "primeflex": "1.0.0-rc.1",
    "primeicons": "^2.0.0",
    "primeng": "7.1.3",
    "quill": "^1.3.6",
    "rxjs": "^6.6.7",
    "rxjs-compat": "^6.0.0-rc.0",
    "tslib": "^1.10.0",
    "web-animations-js": "^2.3.2",
    "zone.js": "~0.10.2"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "~0.901.15",
    "@angular/compiler-cli": "^9.1.13",
    "@fullcalendar/core": "4.1.0",
    "@fullcalendar/daygrid": "4.1.0",
    "@fullcalendar/interaction": "4.1.0",
    "@fullcalendar/timegrid": "4.1.0",
    "@types/jasmine": "2.8.3",
    "@types/node": "^12.11.1",
    "angular-ide": "^0.9.17",
    "codelyzer": "^5.1.2",
    "jasmine-core": "~2.5.2",
    "jasmine-spec-reporter": "~3.2.0",
    "karma": "~2.0.0",
    "karma-chrome-launcher": "~2.2.0",
    "karma-cli": "~1.0.1",
    "karma-coverage-istanbul-reporter": "^0.2.0",
    "karma-jasmine": "~1.1.0",
    "karma-jasmine-html-reporter": "^0.2.2",
    "protractor": "~5.4.0",
    "ts-node": "~4.1.0",
    "tslint": "~5.0.0",
    "typescript": "^3.7.5",
    "web-animations-js": "angular/web-animations-js#release_pr208"
  }
}`

Images error link:
https://ibb.co/fDyRfQY
https://ibb.co/c8v7Hf6
https://ibb.co/nkCmG4N

Thakxx!

**Comment by cetincakiroglu** [comment]
You're trying to use Angular 9 with PrimeNG 7.

PrimeNG versions are synced with Angular versions, for example 

Angular 15 -> PrimeNG 15

Angular 16 -> PrimeNG 16

You need to use PrimeNG 9 with Angular 9.

**Reply by eduardoolimpiodev** [reply]
I will test and get back to you

**Reply by eduardoolimpiodev** [reply]
I'm also getting errors even when using version 9. Could you help me?


`{
  "name": "weblims-ng4",
  "version": "5.0.0",
  "license": "MIT",
  "scripts": {
    "ng": "ng",
    "start": "ng serve",
    "build": "ng build",
    "buildHOM": "ng build --prod --env=hom",
    "test": "ng test",
    "lint": "ng lint",
    "e2e": "ng e2e"
  },
  "private": true,
  "dependencies": {
    "@angular/animations": "^9.1.13",
    "@angular/cdk": "^9.2.4",
    "@angular/cli": "^9.1.15",
    "@angular/common": "^9.1.13",
    "@angular/compiler": "^9.1.13",
    "@angular/core": "^9.1.13",
    "@angular/forms": "^9.1.13",
    "@angular/material": "^9.2.4",
    "@angular/platform-browser": "^9.1.13",
    "@angular/platform-browser-dynamic": "^9.1.13",
    "@angular/platform-server": "^9.1.13",
    "@angular/router": "^9.1.13",
    "@types/minimatch": "^3.0.5",
    "chart.js": "2.1.3",
    "classlist.js": "^1.1.20150312",
    "core-js": "^2.4.1",
    "font-awesome": "^4.7.0",
    "fullcalendar": "^3.1.0",
    "jquery": "^3.1.1",
    "jsbarcode": "^3.5.8",
    "moment": "^2.17.1",
    "nanoscroller": "0.8.7",
    "primeflex": "1.0.0-rc.1",
    "primeicons": "^2.0.0",
    "primeng": "9.1.3",
    "quill": "^1.3.6",
    "rxjs": "^6.6.7",
    "rxjs-compat": "^6.0.0-rc.0",
    "tslib": "^1.10.0",
    "web-animations-js": "^2.3.2",
    "zone.js": "~0.10.2"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "~0.901.15",
    "@angular/compiler-cli": "^9.1.13",
    "@fullcalendar/core": "4.1.0",
    "@fullcalendar/daygrid": "4.1.0",
    "@fullcalendar/interaction": "4.1.0",
    "@fullcalendar/timegrid": "4.1.0",
    "@types/jasmine": "2.8.3",
    "@types/node": "^12.11.1",
    "angular-ide": "^0.9.17",
    "codelyzer": "^5.1.2",
    "jasmine-core": "~2.5.2",
    "jasmine-spec-reporter": "~3.2.0",
    "karma": "~2.0.0",
    "karma-chrome-launcher": "~2.2.0",
    "karma-cli": "~1.0.1",
    "karma-coverage-istanbul-reporter": "^0.2.0",
    "karma-jasmine": "~1.1.0",
    "karma-jasmine-html-reporter": "^0.2.2",
    "protractor": "~5.4.0",
    "ts-node": "~4.1.0",
    "tslint": "~5.0.0",
    "typescript": "^3.7.5",
    "web-animations-js": "angular/web-animations-js#release_pr208"
  }
}
`

Error Images Link:
https://ibb.co/ccxwjB9
https://ibb.co/Mk63qQQ
https://ibb.co/WpNr482

**Comment by eduardoolimpiodev** [comment]
I'm also getting errors even when using version 9. Could you help me?

{ "name": "weblims-ng4", "version": "5.0.0", "license": "MIT", "scripts": { "ng": "ng", "start": "ng serve", "build": "ng build", "buildHOM": "ng build --prod --env=hom", "test": "ng test", "lint": "ng lint", "e2e": "ng e2e" }, "private": true, "dependencies": { "@angular/animations": "^9.1.13", "@angular/cdk": "^9.2.4", "@angular/cli": "^9.1.15", "@angular/common": "^9.1.13", "@angular/compiler": "^9.1.13", "@angular/core": "^9.1.13", "@angular/forms": "^9.1.13", "@angular/material": "^9.2.4", "@angular/platform-browser": "^9.1.13", "@angular/platform-browser-dynamic": "^9.1.13", "@angular/platform-server": "^9.1.13", "@angular/router": "^9.1.13", "@types/minimatch": "^3.0.5", "chart.js": "2.1.3", "classlist.js": "^1.1.20150312", "core-js": "^2.4.1", "font-awesome": "^4.7.0", "fullcalendar": "^3.1.0", "jquery": "^3.1.1", "jsbarcode": "^3.5.8", "moment": "^2.17.1", "nanoscroller": "0.8.7", "primeflex": "1.0.0-rc.1", "primeicons": "^2.0.0", "primeng": "9.1.3", "quill": "^1.3.6", "rxjs": "^6.6.7", "rxjs-compat": "^6.0.0-rc.0", "tslib": "^1.10.0", "web-animations-js": "^2.3.2", "zone.js": "~0.10.2" }, "devDependencies": { "@angular-devkit/build-angular": "~0.901.15", "@angular/compiler-cli": "^9.1.13", "@fullcalendar/core": "4.1.0", "@fullcalendar/daygrid": "4.1.0", "@fullcalendar/interaction": "4.1.0", "@fullcalendar/timegrid": "4.1.0", "@types/jasmine": "2.8.3", "@types/node": "^12.11.1", "angular-ide": "^0.9.17", "codelyzer": "^5.1.2", "jasmine-core": "~2.5.2", "jasmine-spec-reporter": "~3.2.0", "karma": "~2.0.0", "karma-chrome-launcher": "~2.2.0", "karma-cli": "~1.0.1", "karma-coverage-istanbul-reporter": "^0.2.0", "karma-jasmine": "~1.1.0", "karma-jasmine-html-reporter": "^0.2.2", "protractor": "~5.4.0", "ts-node": "~4.1.0", "tslint": "~5.0.0", "typescript": "^3.7.5", "web-animations-js": "angular/web-animations-js#release_pr208" } } 

Error Images Link:
https://ibb.co/ccxwjB9
https://ibb.co/Mk63qQQ
https://ibb.co/WpNr482

**Reply by cetincakiroglu** [reply]
Hi,

Screenshots or copy paste error code doesn't help to understand the issue. Please share a stackblitz reproduce of the issue so we can identify the root cause of error.

**Reply by eduardoolimpiodev** [reply]
Criei o repositório com o código, então fica mais fácil para você me ajudar a identificar como resolver esses problemas. Aqui está o link abaixo e muito obrigado.
Assim que você baixar o código, dê-me um OK para que eu possa tornar o repositório privado.

https://github.com/eduardoolimpiodev/TestW

**Comment by eduardoolimpiodev** [comment]


**Reply by eduardoolimpiodev** [reply]
Hello ?

**Reply by cetincakiroglu** [reply]
I've cloned the repo, your project does not start, and the example is broken. Please provide a working example so we can move on.

**Reply by eduardoolimpiodev** [reply]
If I create a repository with version 8 working, could you put it in version 9 for me?

**Comment by eduardoolimpiodev** [comment]
If I create a repository with version 8 working, could you put it in version 9 for me?

**Reply by cetincakiroglu** [reply]
Unfortunately, I won't. It's part of the PRO support plan, you can visit our discord server and seek help from the community.
Discord server: https://discord.gg/gzKFYnpmCY

Closing this thread because:

- Error is not understandable
- Stackblitz example is not provided
- Shared repo is also broken
